src/test/java/com/microsoft/sqlserver/jdbc/TestResource.java

```diff
@@ -63,6 +63,11 @@ protected Object[][] getContents() {
             {"R_ConnectionURLNull", "The connection URL is null."},
             {"R_connectionIsNotClosed", "The connection is not closed."},
             {"R_invalidExceptionMessage", "Invalid exception message"},
+            {"R_invalidClientSecret", "AADSTS7000215: Invalid client secret provided"},
+            {"R_invalidCertFields",
+                    "Error reading certificate, please verify the location of the certificate.signed fields invalid"},
+            {"R_invalidAADPasswordAuth",
+                    "Failed to authenticate the user {0} in Active Directory (Authentication=ActiveDirectoryPassword)"},
```

Comment:
Maybe future-proof this and make it like R_MSALExecution in SQLServerTestResource

```suggestion
            {"R_invalidAADAuth",
                    "Failed to authenticate the user {0} in Active Directory (Authentication={1})."},
```